tests/json/generator.py

```diff
@@ -1,10 +1,27 @@
 import random
 import rstr
 from faker import Faker
+from faker.providers.python import Provider as PythonProvider
 from jsonschema import RefResolver
 from typing import Union, List
 
 
+class JsonProvider(PythonProvider):
+
+    SUPPORTED_JSON_TYPES = ['str', 'str', 'str', 'str', 'float', 'float',
+                            'int', 'int', 'iso8601', 'uri', 'email']
+
+    def jsondict(self, nb_elements=10, variable_nb_elements=True, *value_types):
+        if not value_types:
```

Comment:
This introduces the possibility of the wrong types being sneaked in. I would just outright ban this case:

`assert not value_types, "Custom value types are not support for JSON"`

Reply:
You could do set difference but I just wouldn't go there.

Reply:
I want to leave the option to use this for fuzz testing in the future. The defaults are safe but if the user wants to be more adventurous I don't want to limit them.

Reply:
We don't need this flexibility right now. I am more worried about correctness. Lets keep this simple for now and simply block this case.